src/components/modules/saver.ts

```diff
@@ -36,16 +36,18 @@ export default class Saver extends Module {
      */
     ModificationsObserver.disable();
 
-    blocks.forEach((block: Block) => {
-      chainData.push(this.getSavedData(block));
-    });
-
-    const extractedData = await Promise.all(chainData);
-    const sanitizedData = await Sanitizer.sanitizeBlocks(extractedData);
+    try {
+      blocks.forEach((block: Block) => {
+        chainData.push(this.getSavedData(block));
+      });
 
-    ModificationsObserver.enable();
+      const extractedData = await Promise.all(chainData);
+      const sanitizedData = await Sanitizer.sanitizeBlocks(extractedData);
 
-    return this.makeOutput(sanitizedData);
+      return this.makeOutput(sanitizedData);
+    } finally {
```

Comment:
why isn't there a catch statement for this try block?

Reply:
Because

```
try {
  // ...
} catch (e) {
  throw e;
} finally {
  // ...
}
```

equals

```
try {
  // ...
} finally {
  // ...
}
```